main/src/main/java/org/mobilitydata/gtfsvalidator/table/GeoJsonFileLoader.java

```diff
@@ -0,0 +1,224 @@
+    List<GtfsGeoJsonFeature> features = new ArrayList<>();
+    boolean hasUnparsableFeature = false;
+    try (InputStreamReader reader = new InputStreamReader(inputStream)) {
+      JsonObject jsonObject = JsonParser.parseReader(reader).getAsJsonObject();
+      if (!jsonObject.has("type")) {
+        noticeContainer.addValidationNotice(new MissingRequiredElementNotice(null, "type", null));
+        throw new UnparsableGeoJsonFeatureException("Missing required field 'type'");
+      } else if (!jsonObject.get("type").getAsString().equals("FeatureCollection")) {
+        noticeContainer.addValidationNotice(
+            new UnsupportedGeoJsonTypeNotice(jsonObject.get("type").getAsString()));
+        throw new UnparsableGeoJsonFeatureException("Unsupported GeoJSON type");
+      }
+      JsonArray featuresArray = jsonObject.getAsJsonArray("features");
+      for (int i = 0; i < featuresArray.size(); i++) {
+        JsonElement feature = featuresArray.get(i);
+        GtfsGeoJsonFeature gtfsGeoJsonFeature = extractFeature(feature, noticeContainer, i);
+        hasUnparsableFeature |= gtfsGeoJsonFeature == null;
+        if (gtfsGeoJsonFeature != null) {
+          features.add(gtfsGeoJsonFeature);
+        }
+      }
+    }
+    if (hasUnparsableFeature) {
+      throw new UnparsableGeoJsonFeatureException("Unparsable GeoJSON feature");
+    }
+    return features;
+  }
+
+  public GtfsGeoJsonFeature extractFeature(
+      JsonElement feature, NoticeContainer noticeContainer, int featureIndex) {
+    GtfsGeoJsonFeature gtfsGeoJsonFeature;
+    List<String> missingRequiredFields = new ArrayList<>();
+    String featureId = null;
+    if (feature.isJsonObject()) {
+      JsonObject featureObject = feature.getAsJsonObject();
+      // Handle feature id
+      if (!featureObject.has(GtfsGeoJsonFeature.FEATURE_ID_FIELD_NAME)) {
+        missingRequiredFields.add(
+            GtfsGeoJsonFeature.FEATURE_COLLECTION_FIELD_NAME
+                + '.'
+                + GtfsGeoJsonFeature.FEATURE_ID_FIELD_NAME);
+      } else {
+        featureId = featureObject.get(GtfsGeoJsonFeature.FEATURE_ID_FIELD_NAME).getAsString();
+        if (featureId == null || featureId.isEmpty()) {
+          missingRequiredFields.add(
+              GtfsGeoJsonFeature.FEATURE_COLLECTION_FIELD_NAME
+                  + '.'
+                  + GtfsGeoJsonFeature.FEATURE_ID_FIELD_NAME);
+        }
+      }
+
+      // Handle feature type
+      if (!featureObject.has(GtfsGeoJsonFeature.FEATURE_TYPE_FIELD_NAME)) {
+        missingRequiredFields.add(
+            GtfsGeoJsonFeature.FEATURE_COLLECTION_FIELD_NAME
+                + '.'
+                + GtfsGeoJsonFeature.FEATURE_TYPE_FIELD_NAME);
+      } else if (!featureObject
+          .get(GtfsGeoJsonFeature.FEATURE_TYPE_FIELD_NAME)
+          .getAsString()
+          .equals("Feature")) {
+        noticeContainer.addValidationNotice(
+            new UnsupportedFeatureTypeNotice(
+                featureIndex,
```

Comment:
Should this customer facing info be 1 based?

Reply:
Good point! Thoughts @emmambd ?

Reply:
Isn't common practice in development 0? I'd stick with that, since anyone creating the data in geojson would likely assume the same.

Reply:
yep 0 should be fine.